drivers/watchdog/wdt_renesas_ra.c

```diff
@@ -0,0 +1,332 @@
+			unsigned int delta = period_ms > config->window.max
+						     ? period_ms - config->window.max
+						     : config->window.max - period_ms;
+
+			if (delta < min_delta) {
+				min_delta = delta;
+				best_divisor = divisor;
+				best_timeout = timeout;
+				best_period_ms = period_ms;
+			}
+		}
+	}
+
+	if (min_delta == UINT_MAX) {
+		LOG_ERR("wdt timeout out of range");
+		return -EINVAL;
+	}
+
+	window_start_idx = (config->window.min * 4 + best_period_ms - 1) / best_period_ms;
+	window_end_idx = (config->window.max * 4 + best_period_ms - 1) / best_period_ms;
+
+	if (window_start_lut[window_start_idx] == WDT_WINDOW_INVALID ||
+	    window_end_lut[window_end_idx] == WDT_WINDOW_INVALID) {
+		LOG_ERR("this wdt timeout is not supported");
+		return -ENOTSUP;
+	}
+
+	p_cfg->clock_division = (wdt_clock_division_t)best_divisor;
+	p_cfg->timeout = (wdt_timeout_t)best_timeout;
+	p_cfg->window_start = (wdt_window_start_t)window_start_lut[window_start_idx];
+	p_cfg->window_end = (wdt_window_end_t)window_end_lut[window_end_idx];
+
+	LOG_INF("actual window min = %.2f ms", window_start_idx * best_period_ms * 0.25);
+	LOG_INF("actual window max = %.2f ms", window_end_idx * best_period_ms * 0.25);
+
+	memcpy(&data->timeout, config, sizeof(struct wdt_timeout_cfg));
+
+	return 0;
+}
+
+static int wdt_renesas_ra_setup(const struct device *dev, uint8_t options)
+{
+	struct wdt_renesas_ra_data *data = dev->data;
+	int ret = 0;
+
+	/*
+	 * TODO: WDT must be restarted with wdt_feed call after sleep -> will be added later when PM
+	 * mode is supported
+	 */
+	if ((options & WDT_OPT_PAUSE_IN_SLEEP) != 0) {
+		LOG_ERR("wdt pause in sleep mode not supported");
+		return -ENOTSUP;
+	}
+
+	wdt_renesas_ra_inst_lock(dev);
+
+	if (atomic_test_bit(&data->device_state, WDT_RENESAS_RA_ATOMIC_ENABLE)) {
+		LOG_ERR("wdt has been already setup");
+		ret = -EBUSY;
+		goto end;
+	}
+
+	if (!atomic_test_bit(&data->device_state, WDT_RENESAS_RA_ATOMIC_TIMEOUT_SET)) {
+		LOG_ERR("wdt timeout should be installed before");
+		ret = -EFAULT;
+		goto end;
+	}
+
+	/* Pause watchdog timer when CPU is halted by the debugger. */
+	R_DEBUG->DBGSTOPCR_b.DBGSTOP_WDT = (options & WDT_OPT_PAUSE_HALTED_BY_DBG) != 0 ? 1 : 0;
+
+	if (R_WDT_Open(&data->wdt_ctrl, &data->wdt_cfg) != FSP_SUCCESS) {
+		LOG_ERR("wdt setup failed");
+		ret = -EIO;
+		goto end;
+	}
+
+	if (R_WDT_Refresh(&data->wdt_ctrl) != FSP_SUCCESS) {
+		LOG_ERR("wdt start failed");
+		ret = -EIO;
+		goto end;
+	}
+
+	atomic_set_bit(&data->device_state, WDT_RENESAS_RA_ATOMIC_ENABLE);
+
+end:
+	wdt_renesas_ra_inst_unlock(dev);
+
+	return ret;
+}
+
+static int wdt_renesas_ra_disable(const struct device *dev)
+{
+	struct wdt_renesas_ra_data *data = dev->data;
+
+	if (!atomic_test_bit(&data->device_state, WDT_RENESAS_RA_ATOMIC_ENABLE)) {
+		LOG_ERR("wdt has not been enabled yet");
+		return -EFAULT;
+	}
+
+	LOG_ERR("wdt can not be stopped once it has started");
+	return -EPERM;
```

Comment:
If the hardware does not support disabling the WDT, I suggest make it simple as much as possible.
```suggestion
    LOG_ERR("Disabling WDT is not supported");
    return -ENOTSUP;
```

Reply:
The -ENOTSUP return code is not map with the description of the [wdt_disable](https://docs.zephyrproject.org/latest/doxygen/html/group__watchdog__interface.html#ga32c32cc911e6a0e20cb2dfdd3945be4e). It seems there is a special case for wdt interface to notify the application that this watchdog instance doesn't support a reconfigure before reset (like https://github.com/zephyrproject-rtos/zephyr/blob/c4ebf713b10ce63f44ee3497159c54377ecc839b/drivers/watchdog/wdt_wwdg_stm32.c#L203 or https://github.com/zephyrproject-rtos/zephyr/blob/c4ebf713b10ce63f44ee3497159c54377ecc839b/drivers/watchdog/wdt_nrfx.c#L99)